Sources/LogInOut.php

```diff
@@ -748,7 +748,7 @@ function Logout($internal = false, $redirect = true)
 			$_SESSION['logout_return'] = $_SESSION['logout_url'];
 		}
 		// Setup the return address.
-		else
+		elseif (array_key_exists('old_url', $_SESSION))
```

Comment:
The [SMF coding guidelines](https://wiki.simplemachines.org/smf/Coding_Guidelines#Functions_.26_Language_Constructs) dictate that this should use `isset()`